android/src/main/java/com/joinflux/flux/segment/SegmentPlugin.java

```diff
@@ -26,24 +26,27 @@ public void initialize(PluginCall call) {
             return;
         }
         String key = call.getString("key");
+        Boolean trackLifecycle = call.getBoolean("trackLifecycle", false);
         if (key == null) {
             call.reject("Write key is required to initialize plugin");
             return;
         }
 
         Context context = this.getContext();
-        Builder builder = new Analytics.Builder(context, key);
-        Boolean trackLifecycle = call.getBoolean("trackLifecycle", false);
-        if (trackLifecycle) {
-            builder.trackApplicationLifecycleEvents().experimentalUseNewLifecycleMethods(false);
-        }
-
-        Boolean recordScreenViews = call.getBoolean("recordScreenViews", false);
-        if (recordScreenViews) {
-            builder.recordScreenViews();
-        }
-        initialized = true;
-        implementation.analytics = builder.build();
+        this.getActivity()
+            .runOnUiThread(
+                new Runnable() {
+                    @Override
+                    public void run() {
+                        Builder builder = new Analytics.Builder(context, key);
+                        if (trackLifecycle) {
+                            builder.trackApplicationLifecycleEvents();
+                        }
+                        initialized = true;
+                        implementation.analytics = builder.build();
+                    }
+                }
+            );
```

Comment:
Assuming this was copied from Segment documentation? If not, do we have good reason for this change?

Reply:
This was a mistake, I've reverted this back to the way we had it before (without the need for UI thread). It should be fixed in [73c7b79](https://github.com/joinflux/capacitor-segment/pull/14/commits/73c7b79a7bf6deeac53eba4a7f6691b1247d4e98)